src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -42,7 +39,7 @@
    */
   public List<IexLastTradedPrice> getLastTradedPriceForSymbols(final List<String> symbols) {
     if (CollectionUtils.isEmpty(symbols)) {
-      return Collections.emptyList();
+      throw new BadArgumentsException("No Stock Symbol Provided.");
```

Comment:
1. The `IexService` doesn't/shouldn't know that the method that called it is speaking HTTP codes. If anything the exception should be more generic.
2. You don't actually need to throw an exception to get the BAD_REQUEST error to show up in the controller. You can use a response entity. https://www.baeldung.com/spring-response-entity

In summary, use a response entity and do any validation of arguments in the Controller, which is aware of the concern that the incoming request is HTTP.

Reply:
Thank you for the discussion - I changed this to be more robust to errors in the Controller and the Service for both the historical prices and the last traded price.